### ISSUE TITLE
Google Chat counter issue [Mac OSX]

### ISSUE THREAD

**Issue by mindingulove** [nativefier/nativefier#1157]
- [ x] I looked at `nativefier --help` and https://github.com/nativefier/nativefier/blob/master/docs/api.md
- [x ] I searched existing issues, open & closed. Yes, my bug is new.
- [x ] I'm using the latest version available at https://github.com/nativefier/nativefier/releases

**Bug description**

When creating google chat app in Mac OS, notification counter in dock won't work properly. Counter displays the " 1 "  and soon disappears after the second message. 

**Steps to reproduce**

nativefier --title-bar-style "hiddenInset" --name "Google Chat" "https://chat.google.com"  --darwin-dark-mode-support  --single-instance   --icon "your-path" --inject "your-path" --honest --widevine --browserwindow-options '{ "fullscreenable": "true", "simpleFullscreen": "false" }' --disable-dev-tools --tray start-in-tray --icon "your path"  --internal-urls "chat\.google\.com" --single-instance --file-download-options '{"showBadge": true}' --counter --verbose

**LOG:**

Running in verbose mode! This will produce a mountain of logs and is recommended only for troubleshooting or if you like Shakespeare.

ATTENTION: Using the **unofficial** Electron from castLabs 
It implements Google's Widevine Content Decryption Module (CDM) for DRM-enabled playback. 
Simply abort & re-run without passing the widevine flag to default to 12.0.4

Performing async options post-processing.
Inferring user agent for electron 12.0.4-wvvmp / darwin
Grabbing electron<->chrome versions file from https://atom.io/download/atom-shell/index.json
Got icon from options. Using it, no inferring needed
Got name Google Chat from options. No inferring needed
Sanitized filename for Google Chat : Google Chat
Unable to infer chrome version for user agent, using 89.0.4389.114
Given chrome 89.0.4389.114 on darwin, using user agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.114 Safari/537.36

Preparing Electron app...
Copying electron app from /opt/homebrew/Cellar/nativefier/43.0.2/libexec/lib/node_modules/nativefier/app to /var/folders/tf/bqs_qxsj699bcb4k4mjvml4w0000gp/T/nativefier-21-18-8-app--22602-IGvXs7Xr02eK
Writing app config to /var/folders/tf/bqs_qxsj699bcb4k4mjvml4w0000gp/T/nativefier-21-18-8-app--22602-IGvXs7Xr02eK/nativefier.json
Copying 1 files to inject in app.
Copying injection file "/Users/jaymeeduardo/Desktop/site.css" to "/var/folders/tf/bqs_qxsj699bcb4k4mjvml4w0000gp/T/nativefier-21-18-8-app--22602-IGvXs7Xr02eK/inject/inject.css"
Updating /var/folders/tf/bqs_qxsj699bcb4k4mjvml4w0000gp/T/nativefier-21-18-8-app--22602-IGvXs7Xr02eK/package.json 'name' field to google-chat-nativefier-c31d42

Converting icons...
Building for macOS and icon is already a .icns, no conversion needed
Copying icons if necessary
No copying necessary on macOS; aborting

Packaging... This will take a few seconds, maybe minutes if the requested Electron isn't cached yet...
  electron-packager Electron Packager 15.2.0
  electron-packager Node v16.0.0
  electron-packager Host Operating system: darwin 20.4.0 (arm64) +0ms
  electron-packager Packager Options: {"arch":"arm64","asar":false,"darwinDarkModeSupport":true,"dir":"/var/folders/tf/bqs_qxsj699bcb4k4mjvml4w0000gp/T/nativefier-21-18-8-app--22602-IGvXs7Xr02eK","electronVersion":"12.0.4-wvvmp","icon":"/Users/jaymeeduardo/electron.icns","name":"Google Chat","out":"/Users/jaymeeduardo","overwrite":true,"platform":"darwin","targetUrl":"https://chat.google.com/","tmpdir":false,"win32metadata":{"ProductName":"Google Chat","InternalName":"Google Chat","FileDescription":"Google Chat"}} +0ms
  electron-packager Target Platforms: darwin +0ms
  electron-packager Target Architectures: arm64 +0ms
  electron-packager Inferring appVersion from version in /var/folders/tf/bqs_qxsj699bcb4k4mjvml4w0000gp/T/nativefier-21-18-8-app--22602-IGvXs7Xr02eK/package.json +0ms
  electron-packager Application name: Google Chat +4ms
  electron-packager Target Electron version: 12.0.4-wvvmp +0ms
  electron-packager Ignored path regular expressions: [
  '/package-lock\\.json$',
  '/yarn\\.lock$',
  '/\\.git($|/)',
  '/node_modules/\\.bin($|/)',
  '\\.o(bj)?$'
] +0ms
  electron-packager Downloading Electron with options {"platform":"darwin","arch":"arm64","version":"12.0.4-wvvmp","artifactName":"electron"} +0ms
Packaging app for platform darwin arm64 using electron v12.0.4-wvvmp
  electron-packager Creating /Users/jaymeeduardo/darwin-arm64-template +4ms
  electron-packager Extracting /Users/jaymeeduardo/Library/Caches/electron/44b8c05f55aec6e3ab20af3a24f6e5dd5b4c13a70158a0616f724670dc740cd0/electron-v12.0.4-wvvmp-darwin-arm64.zip to /Users/jaymeeduardo/darwin-arm64-template +1ms
  electron-packager Initializing app in /Users/jaymeeduardo/Google Chat-darwin-arm64 from /Users/jaymeeduardo/darwin-arm64-template template +0ms
  electron-packager Ignored paths based on the out param: [
  '/Users/jaymeeduardo/Google Chat-darwin-x64',
  '/Users/jaymeeduardo/Google Chat-darwin-arm64',
  '/Users/jaymeeduardo/Google Chat-linux-ia32',
  '/Users/jaymeeduardo/Google Chat-linux-x64',
  '/Users/jaymeeduardo/Google Chat-linux-armv7l',
  '/Users/jaymeeduardo/Google Chat-linux-arm64',
  '/Users/jaymeeduardo/Google Chat-linux-mips64el',
  '/Users/jaymeeduardo/Google Chat-mas-x64',
  '/Users/jaymeeduardo/Google Chat-mas-arm64',
  '/Users/jaymeeduardo/Google Chat-win32-ia32',
  '/Users/jaymeeduardo/Google Chat-win32-x64',
  '/Users/jaymeeduardo/Google Chat-win32-arm64'
] +1s
  electron-packager Copying icon "/Users/jaymeeduardo/electron.icns" to app's Resources as "electron.icns" +0ms
  electron-packager Renaming Electron to Google Chat in /Users/jaymeeduardo/Google Chat-darwin-arm64/Electron.app/Contents/MacOS +24ms
  electron-packager Renaming Electron Helper to Google Chat Helper in /Users/jaymeeduardo/Google Chat-darwin-arm64/Electron.app/Contents/Frameworks/Electron Helper.app/Contents/MacOS +1ms
  electron-packager Renaming Electron Helper (Renderer) to Google Chat Helper (Renderer) in /Users/jaymeeduardo/Google Chat-darwin-arm64/Electron.app/Contents/Frameworks/Electron Helper (Renderer).app/Contents/MacOS +0ms
  electron-packager Renaming Electron Helper (Plugin) to Google Chat Helper (Plugin) in /Users/jaymeeduardo/Google Chat-darwin-arm64/Electron.app/Contents/Frameworks/Electron Helper (Plugin).app/Contents/MacOS +0ms
  electron-packager Renaming Electron Helper (GPU) to Google Chat Helper (GPU) in /Users/jaymeeduardo/Google Chat-darwin-arm64/Electron.app/Contents/Frameworks/Electron Helper (GPU).app/Contents/MacOS +0ms
  electron-packager Renaming Electron Helper.app to Google Chat Helper.app in /Users/jaymeeduardo/Google Chat-darwin-arm64/Electron.app/Contents/Frameworks +0ms
  electron-packager Renaming Electron Helper (Renderer).app to Google Chat Helper (Renderer).app in /Users/jaymeeduardo/Google Chat-darwin-arm64/Electron.app/Contents/Frameworks +0ms
  electron-packager Renaming Electron Helper (Plugin).app to Google Chat Helper (Plugin).app in /Users/jaymeeduardo/Google Chat-darwin-arm64/Electron.app/Contents/Frameworks +0ms
  electron-packager Renaming Electron Helper (GPU).app to Google Chat Helper (GPU).app in /Users/jaymeeduardo/Google Chat-darwin-arm64/Electron.app/Contents/Frameworks +0ms

Finalizing build...
App built to /Users/jaymeeduardo/Google Chat-darwin-arm64, move to wherever it makes sense for you and run the app bundle.

**Expected behavior**

Counter increasing and displaying actual messages number

**Actual behavior**

Counter displays "one message" and disappears 

**Context**

 - Nativefier: (12.0.4)
 - Npm: (7.11.1)
 - OS: Mac OS Big Sur 11.3 Macbook M1
 -

**Comment by mindingulove** [comment]
Here is the app compiled for Mac OS ARM. If someone could reproduce the bug. 
[https://drive.google.com/file/d/19K5gcJQtOVZX59yMkoVSsNU7fkV9E_-K/view?usp=sharing](url)

**Comment by TheCleric** [comment]
I just tried this locally and here's what I encountered:

I did not get a counter on my icon as described. However it doesn't appear as if Google Chat gives up this information readily. Normally for the counter, we're relying on something that looks like a counter in the title of the page. For example in GMail, the title of the page is `Inbox (1) - ...` when you have 1 unread email. I do not see this happening with Chat, which is why the counter doesn't seem to work.
